src/Illuminate/Cookie/CookieJar.php

```diff
@@ -150,7 +150,11 @@ public function queue(...$parameters)
             $this->queued[$cookie->getName()] = [];
         }
 
-        $this->queued[$cookie->getName()][$cookie->getPath()] = $cookie;
+        if (! isset($this->queued[$cookie->getDomain()])) {
+            $this->queued[$cookie->getDomain()] = [];
+        }
```

Comment:
Hey @Sajid-al-islam !
I think the selected part is incorrect, as the `domain` is not in first place inside the queued array (`name` is instead).

As the domain is the last part, it could be something like this (checking if path is set and then checking if domain is set):
```suggestion
        if (! isset($this->queued[$cookie->getName()][$cookie->getPath()])) {
            $this->queued[$cookie->getName()][$cookie->getPath()] = [];
        }
        
        if (! isset($this->queued[$cookie->getName()][$cookie->getPath()][$cookie->getDomain()])) {
            $this->queued[$cookie->getName()][$cookie->getPath()][$cookie->getDomain()] = [];
        }
```

Which then would lead to the domain and array key being `null` if no custom domain has been set, as `null` is the default value for the domain in this class. Same for path, if no path was set. `null` as array key is valid, so everything without an explicit path/domain will also still be valid.

Also, the unqueue method below in this class could be updated to have the domain included, to be able to remove cookies also based on the domain from the queue.

What do you think about that?

Have a blessed day!

Reply:
Yes, I think you are right, let me fix it, and for unqueue method, I did not notice that :)

let me add this in unqueue method too, thanks